modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPInstanceLocalServiceImpl.java

```diff
@@ -430,26 +430,23 @@ public CPInstance addOrUpdateCPInstance(
 			int discontinuedDateYear, ServiceContext serviceContext)
 		throws PortalException {
 
-		if (Validator.isBlank(externalReferenceCode)) {
-			externalReferenceCode = null;
-		}
-		else {
-			CPInstance cpInstance = cpInstancePersistence.fetchByC_ERC(
-				serviceContext.getCompanyId(), externalReferenceCode);
+		CPInstance cpInstance = cpInstancePersistence.fetchByCPDI_SKU(
```

Comment:
I don't think you should do this change

Reply:
@andrea-ale-sbarra , if we keep the default logic, our post call will always fail to do an upsert because we are not including a externalReferenceCode. Is our intended behavior to force an externalReferenceCode to be used always in the API?

Reply:
How would you update the SKU via ERC then? I don't get it